src/typing/flow_js.ml

```diff
@@ -5710,6 +5714,9 @@ let rec __flow cx ((l: Type.t), (u: Type.use_t)) trace =
       ) call_args_tlist;
       rec_flow_t cx trace (l, call_tout)
 
+    | FunProtoApplyT (lreason, _), BindT (_, _, { call_this_t; call_tout; _ }, _) ->
+      rec_flow_t cx trace (FunProtoApplyT (lreason, Some call_this_t), call_tout)
```

Comment:
This doesn't look right. Look at the `FunT _, BindT _` rule for guidance.

1. The `call_this_t` field of the `BindT` use is the this type corresponding to the call of bind. E.g., `x` in `x.bind(...)`. You want the first argument instead.
2. Looking at `FunT _, BindT _` as a guide, you'll notice some complexity missing from here. Specifically, the constraints involving `ResolveSpreadsToMultiflowPartial`. This is needed to figure out the full argument list in the presence of spread arguments.

Reply:
What first argument? This is forwarding From `x.bind(...)` to `$Function$Prototype$Apply<typeof x>`

Reply:
If you write a test, I think you will see what I mean. You want the first argument passed to the bind call, which is not `call_this_t`.

Reply:
Ah, sorry, my mistake -- I was confusing BindT with FunProtoBindT. You're right that `call_this_t` is the first parameter.

You do still need to do something with `call_args_tlist`, though.

Reply:
You mean something like this?
```js
const appliedFn = Function.apply.bind(x); // ok
const reappliedFn = appliedFn.bind(y); // wrong
reappliedFn(null, [""]); // no error
```

Reply:
@samwgoldman I think you was right 😀

Reply:
@samwgoldman would this mean adding second argument with applied parameters? I think one argument is already too much